AsyncDisplayKit/ASViewController.mm

```diff
@@ -132,4 +142,70 @@ - (ASInterfaceState)interfaceState
   return _node.interfaceState;
 }
 
+#pragma mark - ASEnvironmentTraitCollection
+
+- (ASEnvironmentTraitCollection)displayTraitsForTraitCollection:(UITraitCollection *)traitCollection
+{
+  if (self.overrideDisplayTraitsWithTraitCollection) {
+    ASTraitCollection *asyncTraitCollection = self.overrideDisplayTraitsWithTraitCollection(traitCollection);
+    asyncTraitCollection.isMutable = NO;
+    return [asyncTraitCollection environmentTraitCollection];
+  }
+  
+  ASEnvironmentTraitCollection asyncTraitCollection = ASEnvironmentTraitCollectionFromUITraitCollection(traitCollection);
+  asyncTraitCollection.displayContext = _displayTraitsContext;
+  return asyncTraitCollection;
+}
+
+- (ASEnvironmentTraitCollection)displayTraitsForWindowSize:(CGSize)windowSize
+{
+  if (self.overrideDisplayTraitsWithWindowSize) {
+    ASTraitCollection *traitCollection = self.overrideDisplayTraitsWithWindowSize(windowSize);
+    traitCollection.isMutable = NO;
+    return [traitCollection environmentTraitCollection];
+  }
+  return self.node.environmentState.traitCollection;
```

Comment:
@maicki had the idea to expose the trait collection as ASTraitCollection to the user, but store it as a struct behind the scenes. When providing a custom trait collection the user overrides one of two blocks, both of which return ASTraitCollection* not the struct.

I guess my point is that we are trying to make it difficult for the user to get to the struct trait collection since most obj-c developers are more familiar using classes. By adding a getter to the struct traits we kind of break that.